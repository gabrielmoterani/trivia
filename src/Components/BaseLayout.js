import React from 'react';
import { PropTypes } from 'prop-types';
import { StyleSheet } from 'react-native';
import { Layout, Text } from 'react-native-ui-kitten';
import i18n from '../Locales/i18n';

function BaseLayout(props) {
  return (
    <Layout style={Style.container}>
      <Layout style={Style.title}>
        <Text category="h1">trivia</Text>
      </Layout>
      <Layout style={Style.main}>
        {props.children}
      </Layout>
      <Layout style={Style.footer}>
        <Text category="c1">{i18n.t('general.footer')}</Text>
      </Layout>
    </Layout>
  );
}

const Style = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 30,
    marginBottom: 15,
  },
  title: {
    flex: 0.1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',

  },
  main: {
    flex: 0.85,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 10,
  },
  footer: {
    flex: 0.05,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

BaseLayout.propTypes = {
  children: PropTypes.array.isRequired,
};

export default BaseLayout;
