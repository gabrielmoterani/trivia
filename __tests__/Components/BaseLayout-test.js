import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import BaseLayout from '../../src/Components/BaseLayout';

describe('Test BaseLayout component', () => {
  it('renders correctly', () => {
    const renderResult = shallow(
      <BaseLayout>
        <Text>This is a test!</Text>
      </BaseLayout>
    );

    expect(renderResult).toMatchSnapshot();
  });
});
