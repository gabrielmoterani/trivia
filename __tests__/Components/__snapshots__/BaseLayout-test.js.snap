// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BaseLayout component renders correctly 1`] = `
<LayoutComponent
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 15,
      "marginTop": 30,
    }
  }
>
  <LayoutComponent
    style={
      Object {
        "alignItems": "center",
        "flex": 0.1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <TextComponent
      category="h1"
    >
      trivia
    </TextComponent>
  </LayoutComponent>
  <LayoutComponent
    style={
      Object {
        "alignItems": "center",
        "flex": 0.85,
        "justifyContent": "center",
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <Text>
      This is a test!
    </Text>
  </LayoutComponent>
  <LayoutComponent
    style={
      Object {
        "alignItems": "center",
        "flex": 0.05,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <TextComponent
      category="c1"
    >
      made with ❤️by @rafaismyname
    </TextComponent>
  </LayoutComponent>
</LayoutComponent>
`;
